src/site/views/ticketpurchase/tmpl/summary.php

```diff
@@ -8,85 +8,124 @@
 		break;
 	}
 }
 
-if ($ticket == null)
-{
+if ($ticket == null) {
 	$app->enqueueMessage('Purchase failed because the session expired - please try again.', 'Error');
 	$app->redirect(JRoute::_('index.php?option=com_swa&view=ticketpurchase'));
 }
 ?>
 
 <script type="text/javascript" xmlns="http://www.w3.org/1999/html">
-	jQuery(document).ready(function () {
-
+	jQuery(document).ready(function() {
+		$validAddon = true;
+		$cardDetailsComplete = false;
+		$stripeCardErrorMsg = ""
+		$totalDiv = document.getElementById("payment-total");
 		$addons = jQuery('.swa-addon');
-		if ($addons.length > 0) {
-			$qtySelectors = jQuery('.swa-qty-selector');
-			$optionSelectors = jQuery('.swa-option-selector');
 
-			// disable the stripe button
-			$stripeBtn = jQuery('#stripe-button');
-			$stripeBtn.prop('disabled', true);
-
-			// disable and hide all option selectors
-			$optionSelectors.prop('disabled', true);
-			$optionSelectors.parent().prop('hidden', true);
-
-			// define function to calculate the total ticket price
-			$totalTicketPrice = function () {
-				$ticketPrice = parseFloat(jQuery(".swa-ticket").attr('data-price'));
-
-				$totalAddonsPrice = 0;
-				$addons.each(function (i, obj) {
-					$addonQty = parseInt(obj.value);
-					if ($addonQty > 0) {
-						$addonPrice = parseFloat(obj.getAttribute('data-price'));
-						$totalAddonsPrice += $addonPrice * $addonQty
-					}
-				});
+		// define function to enable/disable stripe button
+		setStripeButtonStatus = function() {
+			if (!$validAddon) {
+				document.querySelector("#stripe-button").disabled = true;
+				$ErrorMsg = "Invalid addon selection";
+			} else if (!$cardDetailsComplete) {
+				document.querySelector("#stripe-button").disabled = true;
+				// document.querySelector("#stripe-button").disabled = event.empty;
+				$ErrorMsg = $stripeCardErrorMsg;
+			} else {
+				document.querySelector("#stripe-button").disabled = false;
+				$ErrorMsg = ""
+			}
+			document.querySelector("#card-error").textContent = $ErrorMsg;
+			document.querySelector("#card-error").color = '#fa755a';
+		}
 
-				$totalPrice = $ticketPrice + $totalAddonsPrice;
-				$totalPrice = $totalPrice.toFixed(2);
-				// stripe amount is in pence
-				$stripeBtn.attr('data-amount', $totalPrice * 100);
-				jQuery('#stripe-button span')[0].innerHTML = "Pay £" + $totalPrice + " now!";
+		// define function to calculate the total ticket price
+		$updateTicketPrice = function() {
+			$ticketPrice = parseFloat(jQuery(".swa-ticket").attr('data-price'));
+			$addonsArray = $generateAddonsArray()
+
+			$totalAddonsPrice = 0;
+			$addonsArray.forEach(function(addon) {
+				$addonQty = addon.qty;
+				if ($addonQty > 0) {
+					$addonPrice = addon.price;
+					$totalAddonsPrice += $addonPrice * $addonQty
+				}
+			});
 
-			};
+			$totalPrice = $ticketPrice + $totalAddonsPrice;
+			$totalPrice = $totalPrice.toFixed(2);
+			// stripe amount is in pence
+			$totalDiv.innerHTML = "Total: " + $totalPrice + " GBP";
```

Comment:
As per previous comment, only round ticket price when displaying to the user:
```suggestion
			$totalPrice = $ticketPrice + $totalAddonsPrice;
			$totalDiv.innerHTML = "Total: " + $totalPrice.toFixed(2) + " GBP";
```